package.json

```
@@ -22,10 +22,17 @@
   "exports": {
     ".": {
       "import": "./lib/index.mjs",
-      "require": "./lib/index.js"
+      "require": "./lib/index.js",
+      "types": "./types/index.d.ts"
+    },
+    "./lib/*": {
+      "default": "./lib/*.js",
+      "types": "./types/*.d.ts"
+    },
+    "./lib/errors": {
+      "default": "./lib/errors/index.js",
+      "types": "./types/errors/index.d.ts"
```

```suggestion
      "types": "./types/errors/index"
```
This would probably also need to be without the extension as mentioned in #14610

---

i see the typescript [package-json-exports-imports-and-self-referencing](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7-beta/#package-json-exports-imports-and-self-referencing) docs, the `types` always with `.d.ts`